src/firmware/shared/circular_buffer.h

```diff
@@ -0,0 +1,54 @@
+#include <stdbool.h>
+#include <stdlib.h>
+
+/**
+ * This struct represents the circular buffer properties
+ */
+typedef struct CircularBuffer CircularBuffer_t;
+
+/**
+ * Initialize the circular_buffer struct members
+ * @param  size Size of the circular_buffer to set
+ *
+ * @return A pointer to the created circular_buffer, ownership is given to the caller
+ */
+CircularBuffer_t *circular_buffer_create(size_t size);
+
+/**
+ * Destroy the circular_buffer, freeing any memory allocated for it
+ *
+ * NOTE: This will not destroy the values pointed to by any pointers passed to the
```

Comment:
nit: I know this probably came from `FirmwareWorld` but it's not relevant here since there's no pointers passed on construction. We can remove this.

Reply:
Removed `circular_buffer_destroy()`

Reply:
Sorry I should have been more clear. I just meant the comment about destroying pointers passed to the `create` function. The `destroy` function is still needed.

Reply:
Removed comment and re-added the `destroy` function.